### PR TITLE
Maintenance fixes

### PR DESCRIPTION
**Pull request by maartenvanvliet**
I found a couple of small improvements in the code base

 * Remove unused requires
 * Remove unused variables that exposed bugs in type imports/assertion helpers
 * Remove unused files
 * Rename test file so it gets run

### CODE REVIEW

File: lib/absinthe/pipeline/error_result.ex [hunk 1]
```diff
@@ -1,20 +0,0 @@
-defmodule Absinthe.Pipeline.ErrorResult do
```
Comment (binaryseed) [inline lib/absinthe/pipeline/error_result.ex:1]: I wonder if some users are in fact using this even if Absinthe internals aren't using it... It was part of the docs

File: lib/absinthe/phase/schema/type_imports.ex [hunk 2]
```diff
@@ -37,13 +37,14 @@ defmodule Absinthe.Phase.Schema.TypeImports do
...
 
-    case Keyword.fetch(opts, :only) do
-      {:ok, selections} ->
-        Enum.filter(types, &(&1.identifier in selections))
-
-      _ ->
-        types
-    end
+    types =
+      case Keyword.fetch(opts, :only) do
```
Comment (binaryseed) [inline lib/absinthe/phase/schema/type_imports.ex:41]: No test covered this feature? It seems like this one was covering it:

https://github.com/absinthe-graphql/absinthe/blob/master/test/absinthe/schema/notation/experimental/import_fields_test.exs#L100

File: lib/absinthe/schema/rule.ex [hunk 3]
```diff
@@ -1,40 +0,0 @@
-defmodule Absinthe.Schema.Rule do
```
Comment (binaryseed) [inline lib/absinthe/schema/rule.ex:1]: @benwilson512 Was this leftover from before Schema phases?

**Review by binaryseed — COMMENTED** [review]
Thanks!

File: lib/absinthe/schema/rule.ex (hunk 3 above)
Comment (benwilson512) [reply lib/absinthe/schema/rule.ex:1]: Yes, it can be safely deleted.

File: lib/absinthe/phase/schema/type_imports.ex (hunk 2 above)
Comment (maartenvanvliet) [reply lib/absinthe/phase/schema/type_imports.ex:41]: This test https://github.com/absinthe-graphql/absinthe/blob/master/lib/absinthe/phase/schema/type_imports.ex covers the :only/:except logic for type imports, but the test only asserts that :only/:except is set correctly, not whether the actual import happens. I've added assertions that should check this.

File: lib/absinthe/pipeline/error_result.ex (hunk 1 above)
Comment (maartenvanvliet) [reply lib/absinthe/pipeline/error_result.ex:1]: Happy to leave this in, but easy enough for users to put back if they depend on it.

File: lib/absinthe/phase/schema/type_imports.ex (hunk 2 above)
Comment (maartenvanvliet) [reply lib/absinthe/phase/schema/type_imports.ex:41]: Also added some docs for this functionality and added import_types/2 to locals_without_parens

File: lib/absinthe/pipeline/error_result.ex (hunk 1 above)
Comment (binaryseed) [reply lib/absinthe/pipeline/error_result.ex:1]: Let's move forward and address it if anyone reports any issues

File: lib/absinthe/phase/schema/type_imports.ex (hunk 2 above)
Comment (binaryseed) [reply lib/absinthe/phase/schema/type_imports.ex:41]: Great!

**Pull request merged**